From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Bacteriawa <A3167717663@hotmail.com>
Date: Mon, 4 May 2026 16:29:21 +0800
Subject: [PATCH] Petal: Reduce sensor work

Co-authored by: peaches94 <peachescu94@gmail.com>
As part of: Petal (https://github.com/Bloom-host/Petal/blob/cc691540fb48240f38b376f3d94c8b0db2b60d99/patches/server/0005-feat-reduce-sensor-work.patch)
Licensed under: GPL-3.0 (https://github.com/Bloom-host/Petal/blob/cc691540fb48240f38b376f3d94c8b0db2b60d99/LICENSE)

diff --git a/net/minecraft/world/entity/Mob.java b/net/minecraft/world/entity/Mob.java
index 8a62960aded1189e62c13ef9556b3a8c9b709e76..6a179d9268b4a82eb7721b2cb6081eb63b4ce455 100644
--- a/net/minecraft/world/entity/Mob.java
+++ b/net/minecraft/world/entity/Mob.java
@@ -822,12 +822,12 @@ public abstract class Mob extends LivingEntity implements Targeting, EquipmentUs
             return;
         }
         // Paper end - Allow nerfed mobs to jump and float
+        int i = this.tickCount + this.getId(); // Mint - Petal - Reduce sensor work - Move up
         ProfilerFiller profiler = Profiler.get();
         profiler.push("sensing");
-        this.sensing.tick();
+        if (i % dev.bacteriawa.mint.config.modules.optimizations.PetalReduceSensorWorkConfig.delayTicks == 0 || !dev.bacteriawa.mint.config.modules.optimizations.PetalReduceSensorWorkConfig.enabled) this.sensing.tick(); // Mint - Petal - Reduce sensor work
         profiler.pop();
-        int idBasedTickCount = this.tickCount + this.getId();
-        if (idBasedTickCount % 2 != 0 && this.tickCount > 1) {
+        if (i % 2 != 0 && this.tickCount > 1) {
             profiler.push("targetSelector");
             this.targetSelector.tickRunningGoals(false);
             profiler.pop();
